### ISSUE TITLE
MSB Fish Data Migration

### ISSUE THREAD

**Issue by campmlc** [ArctosDB/arctos#767]
Bulkload of MSB Fish database

**Comment by campmlc** [comment]
Upload lookup table for MSB Fish catalog numbers to replace exisiting MSB/DGR numbers.

**Comment by campmlc** [comment]
Need to add "fin clip" to other IDs.

**Comment by dustymc** [comment]
>Need to add "fin clip" to other IDs.

To what does this apply? If it's an individual from a lot (which needs distinguished as an individual for eg, tracking to genbank), then this should go to part_attribute "other ID."

It's a fairly sloppy system with little to no controls, but it's the only way to deal with not cataloging the things which are studied and cited.

**Comment by campmlc** [comment]
I am proposing to add "fin clip" or "fin clip number", if preferable, to
other identifiers for a single individual, similar to "toe clip number"
which is listed in this way. If the individual with fin clip is part of a
lot, then the number could go in part attributes. But in general, I am
trying to move all members of a lot that need to be distinguished for
tracking purposes to new catalog numbers for GenBank etc.

**Comment by dustymc** [comment]
If this is "unique number that gets clipped to convenient appendage" then it's not comparable to "toe clip number" (but might be the same thing as some other ID type?). If it's a mark-recapture technique or similar (very low-scale unique marking of individuals) then maybe it is the same concept.

If a proposed new value (anywhere) is very similar to some existing value, we should try to combine them - and document in both why we can't if we can't. 

Doing one thing (identifying an individual) two (or more) ways dependent on something unrelated (whatever arbitrary thing lead to cataloging as a lot or not) is never good data organization and should be avoided if possible. 

If cataloging individuals when they're cited/used is possible, I don't see any problems* here. (And YAY everybody!) If not, then this is probably a good time to address the above.

\* Other than the issue of having hundreds of mostly-indistinguishable ID types for no obvious-to-me benefit, but that's something we need to discuss at the scale of Arctos. Making it a tiny bit worse now doesn't seem like a big deal.

**Comment by campmlc** [comment]
My understanding of fin clip from both talking with Lex and with the fish
biologists at the University of Florida is that the fins are physically cut
according to a specified numbering protocol, exactly like toe clip number.
And that this number is used in identifying individuals in a lot that have
been sampled for genetic material.  Is this correct, Lex?
Mariel

**Comment by campmlc** [comment]
Not sure what is being discussed here but in the MSB Division of Fishes,
small fin clips are typically taken (in the field) from fishes' pelvic,
pectoral or caudal fins.  There is nothing in the *placement or number of
fins clipped* that identifies an individual as in the case of reptile toe
clips.  Mark recapture tags (external or internal-subcutaneous) are
typically used for those types of studies.

So, in order to connect the individual voucher (if collected) to its
genetic sample (fin clip) the voucher and fin clip are assigned a unique
number.  In our collection, that number has been the catalog number- 1 to
whatever. (MSB123456-1, MSB123456-2, etc.) These numbers are physically
attached/included with the voucher and fin clip.

FYI.  Fin clipping fishes for mark-recapture is not used (except for
adipose fin in salmonid) because teleost fins can regenerate/heal so that
the fin clipped area may be indistinguishable.   Let me know if this
answers the question or not.  Lex

On Mon, Oct 19, 2015 at 3:57 PM, Mariel Campbell <campbell@carachupa.org>
wrote:

> My understanding of fin clip from both talking with Lex and with the fish
> biologists at the University of Florida is that the fins are physically cut
> according to a specified numbering protocol, exactly like toe clip number.
> And that this number is used in identifying individuals in a lot that have
> been sampled for genetic material.  Is this correct, Lex?
> Mariel
>
> On Mon, Oct 19, 2015 at 3:45 PM, dustymc <notifications@github.com> wrote:
>
>> If this is "unique number that gets clipped to convenient appendage" then
>> it's not comparable to "toe clip number" (but might be the same thing as
>> some other ID type?). If it's a mark-recapture technique or similar (very
>> low-scale unique marking of individuals) then maybe it is the same concept.
>>
>> If a proposed new value (anywhere) is very similar to some existing
>> value, we should try to combine them - and document in both why we can't if
>> we can't.
>>
>> Doing one thing (identifying an individual) two (or more) ways dependent
>> on something unrelated (whatever arbitrary thing lead to cataloging as a
>> lot or not) is never good data organization and should be avoided if
>> possible.
>>
>> If cataloging individuals when they're cited/used is possible, I don't
>> see any problems* here. (And YAY everybody!) If not, then this is probably
>> a good time to address the above.
>>
>> * Other than the issue of having hundreds of mostly-indistinguishable ID
>> types for no obvious-to-me benefit, but that's something we need to discuss
>> at the scale of Arctos. Making it a tiny bit worse now doesn't seem like a
>> big deal.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/ArctosDB/arctos/issues/767#issuecomment-149356472>.
>>
>
>


-- 
*****************************************************
Alexandra M Snyder
Collections Manager-Fishes
Museum of Southwestern Biology
MSC01-2020

Physical address for FedEx and UPS
Bldg.83  Room 204, 302 Yale Blvd NE
University of New Mexico
Albuquerque NM 87131 USA
PH.505.277.6005

http://msb.unm.edu/

**Comment by campmlc** [comment]
So that is different from what I was told they were doing in Florida -
using the pattern of fin clips to identify a sampled individual within a
lot. So it sounds like the ID "fin clip number" could represent different
things to different collections. There is still the issue of how I load
something that reads:
"fin clip: 76443-1"

Adrienne R. has brought up the issue of eartags as a similar situation. Can
we add "eartag number"? Or does this again just get lumped under "original
identifier"?

**Comment by dustymc** [comment]
The marking thing should be unified under one ID type, whatever we call it. It's used for 3 MSB Mammals (and a fossil horse toe - I'll email the paleo-people....).

http://arctos.database.museum/SpecimenResults.cfm?oidtype=toe%20clip%20number

"ear tag" is redundant with SOMETHING - if the tag was applied by ADF&G, it's their number (and they might have more data) - or can be shoved into "original identifier" if nobody knows where it came from.

If there is or could ever be a "MSB123456-2" then the "fin clip" data belongs in part attributes.

**Comment by campmlc** [comment]
Eartag was (and is still) used for thousands of mark-recapture specimens in
the hantavirus survey and Sevilleta studies; some of those individuals died
and ended up in collections. But without a specific ID type, they are
probably listed just under original identifier. I feel we lose information
by lumping well-recognized ID types. I don't see the drawback to using
"eartag 12345"; it contains more information than "original identifier
12345".

**Comment by dustymc** [comment]
>Eartag was (and is still) used for thousands of mark-recapture specimens in the hantavirus survey and Sevilleta studies; some of those individuals died and ended up in collections. But without a specific ID type, they are probably listed just under original identifier. 

Lead a horse to water etc...

If ya'll want a billion ID types, add them to the code table and use them. (And at some level "you" do want some classification - we used to have a half-dozen or so and split that up after some meeting, I think in ABQ way back when. It's probably a good thing that we did, but I also think we went overboard.) I'll ask you to defend them - you'll need that to write useful definitions anyway - but the AC is the final arbitrator of what ends up in the code tables, not me.

I can advise on some aspects of usability.

If you have specific well-defined options and data entry people follow definitions, great things can happen. I can find what I want, I can talk to your data from my data, I can pull it all and make something bigger than the sum of our data, if ToeClipBank comes online we can plug in with no work, etc. Yay everybody. (Everything with a base_URL - 69 of our 232 current values - works this way, and we do in fact successfully use those for two-way communication in and out of Arctos.)

If your types are used arbitrarily, you end up in a worse place (from a user's perspective) than dumping them all into the same container - now I have to guess what you guessed to find them, or more likely just take everything that might be what I want and go from there. We currently have have 2,996 values that are "1" and 1,454,566 that contain "1" - I'll need the latter if I can't predict your classification or format, and nonresolvable IDs do not tend to have predictable formats. (I'm sure that's the extreme example, but it's still illustrative.) Maybe I can do something useful with the numbers once I've found a specimen, but arbitrary classification is unlikely to help with that either. The definitions and data strongly suggest that arbitrary usage and format is the normal situation for most nonresolvable identifiers, so I'm hesitant to recommend adding even more of them.

It seems fairly likely that someone will bring the Mexican wolf studbook online at some point, and if they do the data you have under that ID will be useful in talking to it. (Some of them - there's inconsistency, but it's cleanable.) I think this is a good approach.

EarTagBank seems less somewhat likely, and I'm guessing there are a LOT of Eartag \#1s out there, so I see very little potential value in separating those out. "New Mexico Department of Game Ear Tag Number" might be more useful.
